### PR TITLE
74348e Add applicant relationship custom page

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

This PR introduces a custom relationship page component for use in form 10-10D. A custom page component was created because this section of the form needed dynamic wording functionality. This component will be added to form 10-10D's Applicant Information section in an upcoming PR. For now, it is just being added without being integrated so that the PR stays within a reasonable size. Additional unit testing is planned in a followup PR.

- Added a custom page for collecting applicant's relationship to sponsor w/ dynamic wording
- Team: IVC-CHAMPVA
- Flipper: NA, not set to go to prod

## Related issue(s)
Part of ongoing work to split up this PR:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/74348

## Testing done

- Manual testing
- Ran `yarn test:coverage-app ivc-champva/10-10D` to verify existing tests run and pass

## Screenshots

|         | Scenario 1 | Scenario 2 |
| ------- | ------ | ----- |
| Desktop |||

## What areas of the site does it impact?

This form only

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

NA

### CODE REVIEW

**Review by balexandr — DISMISSED** [review]
Lgtm

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+      value: 'child',
+    },
+    {
+      label: `${relativeBeingVerb} the ${personTitle}'s caretaker`,
+      value: 'caretaker',
+    },
+    {
+      label: `${`${
+        applicant && !useFirstPerson ? `${applicant} doesn’t` : 'We don’t'
+      }`} have a relationship that’s listed here`,
```
Comment (micahchiang) [inline src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx:52]: Can this be refactored? It doesn't seem like doubly embedded template strings should be used here.

Comment (micahchiang) [reply src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx:52]: Something like this should work: 

```
`${ applicant && !useFirstPerson ? applicant+" doesn't" : "we don't"} have a relationship that's listed here`
```

image

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+              value={option.value}
+              checked={checkValue.relationshipToVeteran === option.value}
+              aria-describedby={
+                checkValue.relationshipToVeteran === option.value
+                  ? option.value
+                  : null
+              }
+            />
+          ))}
+        </VaRadio>
```
Comment (micahchiang) [inline src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx:241]: Can these be converted to v3 by adding the `uswds` prop?

**Review by micahchiang — CHANGES_REQUESTED** [review]
Requesting changes to refactor the multi-leveled template strings. They aren't necessary.

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx (hunk 1 above)
Comment (michaelclement) [reply src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx:52]: Can definitely fix that! I do get a linting error when I use regular string concatenation for that middle part, but getting rid of the outer template works and it looks cleaner. Is this compromise alright?

Screenshot 2024-02-16 at 15 58 00

Screenshot 2024-02-16 at 15 58 17

Comment (micahchiang) [reply src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx]: Thanks @michaelclement - looks much cleaner :+1:

**Review by micahchiang — APPROVED** [review]

**Pull request merged**